sdk/monitor/Microsoft.OpenTelemetry.Exporter.AzureMonitor/src/TelemetryPartA.cs

```diff
@@ -49,6 +52,29 @@ internal static TelemetryItem GetTelemetryItem(Activity activity, string instrum
             return telemetryItem;
         }
 
+        internal static TelemetryItem GetTelemetryItem(LogRecord logRecord, string instrumentationKey)
+        {
+            var name = PartA_Name_Mapping[TelemetryType.Message];
+            var time = logRecord.Timestamp.ToString(CultureInfo.InvariantCulture);
+
+            TelemetryItem telemetryItem = new TelemetryItem(name, time)
+            {
+                InstrumentationKey = instrumentationKey
+            };
+
+            // TODO: I WAS TOLD THIS MIGHT BE CHANGING. IGNORING FOR NOW.
+            //InitRoleInfo(activity);
+            //telemetryItem.Tags[ContextTagKeys.AiCloudRole.ToString()] = RoleName;
+            //telemetryItem.Tags[ContextTagKeys.AiCloudRoleInstance.ToString()] = RoleInstance;
+
+            telemetryItem.Tags[ContextTagKeys.AiOperationId.ToString()] = logRecord.TraceId.ToHexString();
```

Comment:
`ContextTagKeys.AiOperationId.ToString()` - is this a constant thing? Does it create a new string all the time?

Reply:
This is one of the swagger classes. This entire class is a wrapper around known string values. Here, the 'ToString' returns the internal value

https://github.com/Azure/azure-sdk-for-net/blob/9e0bd1168af739274a54b1189be5cc76e6505880/sdk/monitor/Microsoft.OpenTelemetry.Exporter.AzureMonitor/src/Generated/Models/ContextTagKeys.cs#L127